### PR TITLE
MM-36168: Remove mentions to specific tiers

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
This PR updates the user-facing strings to remove any mentions to specific subscription tiers. The copy was suggested by @justinegeffen [in this document](https://docs.google.com/document/d/1gkJemwCDHzlKt2cBD6f2Zvn-mJmhzgOl92MiiW_MrK4/edit), and I adapted it on top of the new one, added in #603.

The only missing mention to Enterprise is from the Export channel button, which is covered by https://mattermost.atlassian.net/browse/MM-36689.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-36168

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: webapp/src/components/backstage/convert_enterprise_notice.tsx [hunk 1]
```diff
@@ -10,14 +10,15 @@ const ConvertEnterpriseNotice = () => (
                 <li>
                     <a
                         href={'https://docs.mattermost.com/install/ee-install.html#converting-team-edition-to-enterprise-edition'}
+
                         target={'_blank'}
                         rel='noreferrer'
                     >
-                        {'Convert to Enterprise Edition.'}
+                        {'Convert to a commercial edition.'}
```
Comment (agarciamontoro) [inline webapp/src/components/backstage/convert_enterprise_notice.tsx:17]: This term was suggested in the SKU Repackaging channel as a collective noun for the non-TE editions (i.e. Starter, Professional and Enterprise): https://community-daily.mattermost.com/private-core/pl/deoqxa3kstfp8k4qkoqnrnyroc

File: server/app/permissions.go [hunk 2]
```diff
@@ -199,7 +199,7 @@ func PlaybookAccess(userID string, playbook Playbook, pluginAPI *pluginapi.Clien
 // checkPlaybookIsNotUsingE20Features features returns a non-nil error if the playbook is using E20 features
 func checkPlaybookIsNotUsingE20Features(playbook Playbook) error {
 	if len(playbook.MemberIDs) > 0 {
-		return errors.Wrap(ErrLicensedFeature, "restrict playbook editing to specific users is a Mattermost Enterprise feature")
+		return errors.Wrap(ErrLicensedFeature, "restricting playbook editing to specific users is not available with your current subscription")
```
Comment (agarciamontoro) [inline server/app/permissions.go:202]: Not sure about the `restricting`, please take a look

Comment (lieut-data) [reply server/app/permissions.go:202]: Nice grammar (and use of the word `gerund` :p)

File: webapp/src/components/backstage/convert_enterprise_notice.tsx [hunk 3]
```diff
@@ -10,14 +10,15 @@ const ConvertEnterpriseNotice = () => (
                 <li>
                     <a
                         href={'https://docs.mattermost.com/install/ee-install.html#converting-team-edition-to-enterprise-edition'}
+
```
Comment (lieut-data) [inline webapp/src/components/backstage/convert_enterprise_notice.tsx:13]: nit: spurious newline?

**Review by lieut-data — APPROVED** [review]
:tada:

File: webapp/src/components/backstage/convert_enterprise_notice.tsx (hunk 3 above)
Comment (agarciamontoro) [reply webapp/src/components/backstage/convert_enterprise_notice.tsx]: Indeed. Removed!

**Review by crspeller — APPROVED** [review]

**Pull request merged**